--recurse-submodules not available in git that ships with RHEL7

Unfortunately, the fix for #122 breaks `check-manifest` on current releases of RHEL as they still ship with git 1.8.

You get a traceback that looks like:

```
E             File "site-packages/check_manifest.py", line 524, in get_vcs_files
E               return canonical_file_list(vcs.get_versioned_files())
E             File "site-packages/check_manifest.py", line 401, in get_versioned_files
E               encoding=self._encoding,
E             File "site-packages/check_manifest.py", line 163, in run
E               (output + stderr).decode(encoding, 'replace'))
E           check_manifest.CommandFailed: ['git', 'ls-files', '-z', '--recurse-submodules'] failed (status 129):
E           error: unknown option `recurse-submodules'
```

--- a/check_manifest.py
+++ b/check_manifest.py
@@ -394,10 +394,14 @@ class Git(VCS):
         # .git can be a file for submodules
         return os.path.exists(os.path.join(location, cls.metadata_name))
 
+    def _has_submodules(cls):
+        return os.path.exists(".gitmodules")
+
     def get_versioned_files(self):
         """List all files versioned by git in the current directory."""
+        extra_args = ["--recurse-submodules"] if self._has_submodules() else []
         output = run(
-            ["git", "ls-files", "-z", "--recurse-submodules"],
+            ["git", "ls-files", "-z"] + extra_args,
             encoding=self._encoding,
         )
         # -z tells git to use \0 as a line terminator; split() treats it as a

